libmachine/provision/ubuntu.go

```diff
@@ -151,18 +164,58 @@ func (provisioner *UbuntuProvisioner) CompatibleWithHost() bool {
 	return provisioner.OsReleaseInfo.Id == "ubuntu"
 }
 
+func (provisioner *UbuntuProvisioner) GetAuthOptions() auth.AuthOptions {
+	return provisioner.AuthOptions
+}
+
 func (provisioner *UbuntuProvisioner) SetOsReleaseInfo(info *OsRelease) {
 	provisioner.OsReleaseInfo = info
 }
 
-func (provisioner *UbuntuProvisioner) GenerateDockerOptions(dockerPort int, authOptions auth.AuthOptions) (*DockerOptions, error) {
-	defaultDaemonOpts := getDefaultDaemonOpts(provisioner.Driver.DriverName(), authOptions)
-	daemonOpts := fmt.Sprintf("--host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:%d", dockerPort)
+func (provisioner *UbuntuProvisioner) GenerateDockerOptions(dockerPort int) (*DockerOptions, error) {
+	var (
+		engineCfg bytes.Buffer
+	)
+
+	driverNameLabel := fmt.Sprintf("provider=%s", provisioner.Driver.DriverName())
+	provisioner.EngineOptions.Labels = append(provisioner.EngineOptions.Labels, driverNameLabel)
+
+	engineConfigTmpl := `
+DOCKER_OPTS='
+-H tcp://0.0.0.0:{{.DockerPort}}
+-H unix:///var/run/docker.sock
+--storage-driver {{.EngineOptions.StorageDriver}}
+--tlsverify
+--tlscacert {{.AuthOptions.CaCertRemotePath}}
+--tlscert {{.AuthOptions.ServerCertRemotePath}}
+--tlskey {{.AuthOptions.ServerKeyRemotePath}}
+{{ range .EngineOptions.Labels }}
```

Comment:
If you wanted to condense these to reduce whitespace in the config you can use it like this (although it would also reduce readability):

```
{{ range .EngineOptions.Labels }}--label {{.}}{{ end}}
```